backend/app/operations_dashboard.py

```
@@ -126,28 +258,168 @@ async def fetch_operations_dashboard(
+         where {visible}
+           and {visible_evidence}
+           and ($5::boolean is false or milestone.case_kind_code = 'external_information')
+        union all
+        select missing.post_id, missing.case_kind_code,
+               missing.milestone_type_code, null, null, null, null, true
+          from operations_case_missing_milestone missing
+          join source_post post on post.post_id = missing.post_id
+         where {visible}
+           and ($5::boolean is false or missing.case_kind_code = 'external_information')
+         order by post_id, case_kind_code, milestone_type_code
+        """,
```

Ruff (0.16.2)

[error] Possible SQL injection vector through string-based query construction

(S608)